src/Core/Plugin.php

```diff
@@ -170,6 +184,24 @@ public static function load($plugin, array $config = [])
     }
 
     /**
+     * Loads the config mappings for plugins.
+     *
+     * The map data in CONFIG/plugins.php map data is used set default paths
+     * for where plugins are located.
+     *
+     * @return void
+     */
+    protected static function _loadConfigMap()
+    {
+        if (!file_exists(CONFIG . 'plugins.php')) {
+            static::$_config = [];
+            return;
+        }
+        $config = include CONFIG . 'plugins.php';
```

Comment:
Would also mean you need composer/composer itself as require in
cakephp/cakephp bringing in a whole bunch of other packages as well. Also
would mean you dont only need composer to install but also to run cakephp.
Op 12 jan. 2015 13:53 schreef "Andy Dawson" <notifications@github.com>:

> In src/Core/Plugin.php
> <https://github.com/cakephp/cakephp/pull/5641#discussion-diff-22787899>:
>
> > @@ -170,6 +184,24 @@ public static function load($plugin, array $config = [])
> >      }
> >
> >      /**
> > +     * Loads the config mappings for plugins.
> > +     *
> > +     * The map data in CONFIG/plugins.php map data is used set default paths
> > +     * for where plugins are located.
> > +     *
> > +     * @return void
> > +     */
> > +    protected static function _loadConfigMap()
> > +    {
> > +        if (!file_exists(CONFIG . 'plugins.php')) {
> > +            static::$_config = [];
> > +            return;
> > +        }
> > +        $config = include CONFIG . 'plugins.php';
>
> I don't think so.
>
> There's vendor/composer/installed.json which does list all installed
> packages and their type - but that's an implementation detail of composer
> itself.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cakephp/cakephp/pull/5641/files#r22787899>.
>